src/diagonal.works/b6/world.go

```diff
@@ -715,6 +745,39 @@ type RelationFeature interface {
 	Covering(coverer s2.RegionCoverer) s2.CellUnion
 }
 
+type CollectionFeature interface {
+	Feature
+	FeatureID() FeatureID
+	CollectionID() CollectionID
+	CollectionProto() pb.CollectionProto
```

Comment:
should i make proto a pointer ? or plot twist u dont think i should use proto at all :)

Reply:
PLOT TWIST! I don't think you should use a proto at all, since depending on them drags a lot of stuff into the interface that I'd prefer to avoid. I think it should be something like:
```
type CollectionFeature interface {
	Feature
	FeatureID() FeatureID
	CollectionID() CollectionID
	Key(i int) interface{}
        Value(i int) interface{}
}
```

We should probably make it conform to the Collection/CollectionIterator interfaces too, but we can add that later.

Reply:
easy ! i ll iterator tmrw